app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -2014,6 +2018,31 @@ class PluginApi {
   registerHashtagSearchParam(param, context, priority) {
     registerHashtagSearchParam(param, context, priority);
   }
+
+  /**
+   * Add a new button to be overlaid ontop of a table appearing in a post.
+   * On wider tables the button will be appending along with the Expand table button.
+   *
+   * Example usage:
+   *
+   *  api.addTableWrapperButton(
+   *    themePrefix("discourse_table_builder.edit.btn_edit"),
+   *    "pencil-alt",
+   *    ["btn-edit-table"],
+   *    generateModalEvent
+   * );
+   *
+   * @param {string} label - An i18n key for the button label
+   * @param {icon} icon - An icon name from fontawesome
+   * @param {array} classes - An array of strings to be added as classes to the button
+   * @param {function} event - A function that returns an event to be dispatched when the button is clicked
+   *
+   */
+  addTableWrapperButton(label, icon, classes, event) {
```

Comment:
Just looking at this from an API perspective, is there a way to target a particular table in a post? It seems like this API will current just add the button to all the tables in the post.

Reply:
Yes, it should be possible through the `event` by getting the element's `data-table-id` attribute like so:
```javascript
const tableId = event.target.getAttribute("data-table-id");
```

It's not immediately, obvious, so I'll be sure to add that to the comment.